### PR TITLE
Added evaluation config under llama3_1 dir, updated registry

### PR DESCRIPTION
**Pull request by ysurs**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [X] other (cleanup)

Please link to any issues this PR addresses. 
https://github.com/pytorch/torchtune/issues/1810

#### Changelog
What are the changes made in this PR?
* Copied evaluation.yaml to llama3_1/ directory
* Updated evaluation.yaml to point to Llama-3.1-8B model instantiations
* Updated the recipe registry class to pick up the new Llama-3.1 config

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [X] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [X] manually run any new or modified recipes with sufficient proof of correctness
- [X] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

Llama 3.1 8B evaluation recipe output:
```
(env) root@035a45fe5d82:~/torchtune# tune run eleuther_eval --config llama3_1/evaluation
INFO:torchtune.utils._logging:Running EleutherEvalRecipe with resolved config:

batch_size: 8
checkpointer:
  _component_: torchtune.training.FullModelHFCheckpointer
  checkpoint_dir: /tmp/Meta-Llama-3.1-8B-Instruct
  checkpoint_files:
  - model-00001-of-00004.safetensors
  - model-00002-of-00004.safetensors
  - model-00003-of-00004.safetensors
  - model-00004-of-00004.safetensors
  model_type: LLAMA3
  output_dir: ./
device: cuda
dtype: bf16
enable_kv_cache: true
limit: null
max_seq_length: 4096
model:
  _component_: torchtune.models.llama3_1.llama3_1_8b
output_dir: ./
quantizer: null
seed: 1234
tasks:
- truthfulqa_mc2
tokenizer:
  _component_: torchtune.models.llama3.llama3_tokenizer
  max_seq_len: null
  path: /tmp/Meta-Llama-3.1-8B-Instruct/original/tokenizer.model

INFO:torchtune.utils._logging:Model is initialized with precision torch.bfloat16.
config.json: 100%|████████████████████████████████████████████████████████████████████████████████████████████| 665/665 [00:00<00:00, 4.81MB/s]
tokenizer_config.json: 100%|█████████████████████████████████████████████████████████████████████████████████| 26.0/26.0 [00:00<00:00, 206kB/s]
vocab.json: 100%|█████████████████████████████████████████████████████████████████████████████████████████| 1.04M/1.04M [00:00<00:00, 1.66MB/s]
merges.txt: 100%|███████████████████████████████████████████████████████████████████████████████████████████| 456k/456k [00:00<00:00, 1.12MB/s]
tokenizer.json: 100%|█████████████████████████████████████████████████████████████████████████████████████| 1.36M/1.36M [00:00<00:00, 6.40MB/s]
model.safetensors: 100%|█████████████████████████████████████████████████████████████████████████████████████| 548M/548M [00:05<00:00, 100MB/s]
generation_config.json: 100%|█████████████████████████████████████████████████████████████████████████████████| 124/124 [00:00<00:00, 1.72MB/s]
README.md: 100%|██████████████████████████████████████████████████████████████████████████████████████████| 9.59k/9.59k [00:00<00:00, 66.1MB/s]
validation-00000-of-00001.parquet: 100%|████████████████████████████████████████████████████████████████████| 271k/271k [00:00<00:00, 11.3MB/s]
Generating validation split: 100%|█████████████████████████████████████████████████████████████████| 817/817 [00:00<00:00, 76056.96 examples/s]
INFO:torchtune.utils._logging:Running evaluation on the following tasks: ['truthfulqa_mc2']
100%|██████████████████████████████████████████████████████████████████████████████████████████████████████| 817/817 [00:00<00:00, 1377.35it/s]
Running loglikelihood requests: 100%|██████████████████████████████████████████████████████████████████████| 5882/5882 [02:35<00:00, 37.81it/s]
INFO:torchtune.utils._logging:Eval completed in 157.77 seconds.
INFO:torchtune.utils._logging:Max memory allocated: 41.92 GB
INFO:torchtune.utils._logging:

|    Tasks     |Version|Filter|n-shot|Metric|   |Value |   |Stderr|
|--------------|------:|------|-----:|------|---|-----:|---|-----:|
|truthfulqa_mc2|      2|none  |     0|acc   |↑  |0.5406|±  | 0.015|
```

### CODE REVIEW

**Review by joecummings — APPROVED** [review]
LGTM! Thanks @ysurs

**Comment by joecummings** [comment]
> Hey @joecummings , can you recommend a slightly tougher issue to work on?
> 
> Would like to level up.

Would you be interested in taking on #2758 (also see #2533)? It's a bit open-ended so please message me on Discord or tag me here if you have any questions!

**Comment by ysurs** [comment]
> > Hey @joecummings , can you recommend a slightly tougher issue to work on?
> > Would like to level up.
> 
> Would you be interested in taking on #2758 (also see #2533)? It's a bit open-ended so please message me on Discord or tag me here if you have any questions!

Sure, thanks for the recommendation, will check this out and reach out to you for more information.
